src/AddressRegistry.Projections.Wms/AddressDetail/AddressDetailProjections.cs

```diff
@@ -0,0 +1,575 @@
+                        address.PositionSpecification = null;
+                        UpdateVersionTimestamp(address, message.Message.Provenance.Timestamp);
+                    },
+                    (previousPositionAddresses, houseNumberLabel) =>
+                    {
+                        foreach (var address in previousPositionAddresses)
+                        {
+                            address.HouseNumber = houseNumberLabel;
+                            UpdateVersionTimestamp(address, message.Message.Provenance.Timestamp);
+                        }
+                    },
+                    default,
+                    ct);
+            });
+
+            When<Envelope<AddressBoxNumberWasChanged>>(async (context, message, ct) =>
+            {
+                await context.FindAndUpdateAddressDetail(
+                    message.Message.AddressId,
+                    item =>
+                    {
+                        item.BoxNumber = message.Message.BoxNumber;
+                        item.LabelType = string.IsNullOrWhiteSpace(message.Message.BoxNumber)
+                            ? WmsAddressLabelType.HouseNumber
+                            : WmsAddressLabelType.BusOrApartmentNumber;
+                        UpdateVersionTimestamp(item, message.Message.Provenance.Timestamp);
+                    },
+                    ct);
+            });
+
+            When<Envelope<AddressBoxNumberWasCorrected>>(async (context, message, ct) =>
+            {
+                await context.FindAndUpdateAddressDetail(
+                    message.Message.AddressId,
+                    item =>
+                    {
+                        item.BoxNumber = message.Message.BoxNumber;
+                        item.LabelType = string.IsNullOrWhiteSpace(message.Message.BoxNumber)
+                            ? WmsAddressLabelType.HouseNumber
+                            : WmsAddressLabelType.BusOrApartmentNumber;
+                        UpdateVersionTimestamp(item, message.Message.Provenance.Timestamp);
+                    },
+                    ct);
+            });
+
+            When<Envelope<AddressBoxNumberWasRemoved>>(async (context, message, ct) =>
+            {
+                await context.FindAndUpdateAddressDetail(
+                    message.Message.AddressId,
+                    item =>
+                    {
+                        item.BoxNumber = null;
+                        item.LabelType = WmsAddressLabelType.HouseNumber;
+                        UpdateVersionTimestamp(item, message.Message.Provenance.Timestamp);
+                    },
+                    ct);
+            });
+        }
+
+        private static bool TryParsePosition(string extendedWkbGeometry, out Point? point)
+        {
+            try
+            {
+                if (string.IsNullOrWhiteSpace(extendedWkbGeometry))
+                    throw new ParseException("Invalid extendedWkbGeometry");
```

Comment:
you can return `null` and crash if the `WKBReader` doesn't work.
Otherwise you swallow the exception while something is very unexpected here.

Reply:
You can also inject the `WKBReader` see: https://github.com/Informatievlaanderen/building-registry/blob/main/src/BuildingRegistry.Projections.Wms/BuildingUnit/BuildingUnitProjections.cs

Reply:
When looking for `TryParsePosition`. It is only used in methods where a position is expected, so null check isn't even needed.